fix(coder): test_composition_completeness_python_fixture_* tests run against consumer repos

## Summary

Two toolkit-internal fixture tests in `test_composition_completeness.py` run when a consumer executes `atdd validate coder`, and fail with assertion errors referencing toolkit test fixtures (e.g., `bad_match/orchestrate_bad`). This is the same class of bug as #272 (fixed in 1.51.1 for `test_structured_logging` dogfooding) — toolkit dogfood tests leak into consumer validation when they should be scoped to the atdd source repo only.

First observed after upgrading `janetbusiness/jel-app` from `atdd==1.49.0` → `atdd==1.53.0`. These failures did not exist in 1.49.0.

## Concrete evidence

Running `atdd validate coder --local` in `janetbusiness/jel-app` against commit `80d16fa6` with `atdd==1.53.0`:

```
_ test_composition_completeness_python_fixture_passes_for_complete_and_partial_features _
[gw1] darwin -- Python 3.14.2 /opt/homebrew/opt/python@3.14/bin/python3.14
/opt/homebrew/lib/python3.14/site-packages/atdd/coder/validators/test_composition_completeness.py:702: in test_composition_completeness_python_fixture_passes_for_complete_and_partial_features
    assert_no_violations(violations)
/opt/homebrew/lib/python3.14/site-packages/atdd/coder/validators/test_composition_completeness.py:649: in assert_no_violations
    assert not violations, "..."
E   AssertionError:
```

```
___ test_composition_completeness_python_fixture_detects_missing_setter_call ___
[gw1] darwin -- Python 3.14.2 /opt/homebrew/opt/python@3.14/bin/python3.14
/opt/homebrew/lib/python3.14/site-packages/atdd/coder/validators/test_composition_completeness.py:715: in test_composition_completeness_python_fixture_detects_missing_setter_call
    assert len(violations) == 1, "expected one composition root violation"
E   AssertionError: expected one composition root violation
E   assert 4 == 1
E    +  where 4 = len([CompositionViolation(spec_id='SPEC-CODER-COMP-0001', feature='bad_match/orchestrate_bad', file='src/application/play_...)])
```

The `bad_match/orchestrate_bad` feature name in the violation report is a dead giveaway that this is a toolkit **fixture** being tested as a regression — it is not present in the consumer repo. The test expects `len(violations) == 1` from a toolkit-side synthetic fixture, but when run against a consumer repo the walker is probably picking up extra violations from the consumer's real code and the fixture assertion fails.

## Why this is the same class as #272

#272 was closed with "scope structured_logging dogfooding to atdd source repo" in PR #275, released as `atdd==1.51.1`. The pattern is identical:

- A toolkit-internal test that validates the validator's own behavior against toolkit fixtures
- The test runs unconditionally when the validator module is imported (pytest discovery)
- Consumer repos installing atdd and running `atdd validate coder` execute the toolkit tests too
- The tests fail because the consumer repo context is not the toolkit source repo

The fix pattern from #272 (scope dogfooding to atdd source repo) should apply here: gate `test_composition_completeness_python_fixture_*` so they only run when the test is executing inside the atdd source repo, or move them out of the shipped package into the atdd source repo's own test suite.

## Impact

- **`janetbusiness/jel-app`'s `validate-coder` now has 2 additional blocker failures** after upgrading from 1.49.0 → 1.53.0, introduced purely by the upgrade
- Upgrade path from 1.49.0 adds these 2 failures even for repos that had no relevant code changes
- Ratchet baselines do not gate these tests — they are always-on assertion tests, so even a repo with zero regressions cannot pass them

## Reproduction

```bash
git clone https://github.com/janetbusiness/jel-app
cd jel-app
git checkout 80d16fa6
pip install atdd==1.53.0 --force-reinstall
atdd validate coder --local 2>&1 | grep -A3 fixture_
```

Expected: fixture tests either skip or do not run in consumer context.
Actual: two fixture tests execute and fail with assertion errors referencing toolkit fixture data (`bad_match/orchestrate_bad`).

## Suggested fix direction

Following the pattern from #272:

1. **Gate the fixture tests to atdd source repo.** Check `REPO_ROOT` (or similar) against a marker file unique to the atdd source repo (e.g., `src/atdd/__init__.py` existence in repo root, or `pyproject.toml` name field). Skip with `pytest.skip` when not in the toolkit source.
2. **Or move the fixture tests out of the shipped package.** If the `test_composition_completeness_python_fixture_*` tests live alongside the validator module in `src/atdd/coder/validators/test_composition_completeness.py`, they ship to consumers. Move them to the atdd repo's top-level `tests/` directory so they don't get installed into `site-packages`.

Option 2 is more robust long-term — dogfood tests shouldn't ship to consumers at all.

## Affected validators

Both are in `test_composition_completeness.py` (lines 702 and 715 per the traceback):

- `test_composition_completeness_python_fixture_passes_for_complete_and_partial_features`
- `test_composition_completeness_python_fixture_detects_missing_setter_call`

Worth auditing other `test_*_fixture_*` patterns in the coder validator tree for the same issue.

## Notes

- #272 was filed for `test_structured_logging` and fixed promptly — thank you.
- This issue is strictly about fixture tests leaking into consumer context. The real `test_composition_completeness_python_live_repo` regression in jel-app is a consumer-side issue and out of scope here.

## References

- #272 — previous dogfood leak, fixed in 1.51.1 via #275
- Reported from: `janetbusiness/jel-app` at `80d16fa6` on `atdd==1.53.0`

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -4,7 +4,7 @@ build-backend = "setuptools.build_meta"
 
 [project]
 name = "atdd"
-version = "1.53.1"
+version = "1.53.2"
 description = "ATDD Platform - Acceptance Test Driven Development toolkit"
 requires-python = ">=3.10"
 readme = "README.md"

--- a/src/atdd/coach/commands/tests/test_is_atdd_source_repo.py
+++ b/src/atdd/coach/commands/tests/test_is_atdd_source_repo.py
@@ -1,0 +1,61 @@
+"""
+Unit tests for `is_atdd_source_repo()` — the dogfood-scope guard.
+
+Regression for #276: fixture-based dogfood tests must be gated by this helper
+so they don't leak into consumer `atdd validate coder` runs. The helper must
+return True only in the atdd source checkout and False in any pip-installed
+/ vendored / consumer-repo layout.
+"""
+from __future__ import annotations
+
+from pathlib import Path
+from unittest.mock import patch
+
+import pytest
+
+from atdd.coach.utils.repo import is_atdd_source_repo, find_repo_root
+
+pytestmark = [pytest.mark.platform]
+
+
+def _make_fake_consumer_repo(root: Path) -> None:
+    (root / ".git").mkdir(parents=True)
+    (root / "pyproject.toml").write_text(
+        '[project]\nname = "consumer-app"\nversion = "0.1.0"\n'
+    )
+
+
+def test_returns_false_when_pkg_in_site_packages(tmp_path, monkeypatch):
+    """pip-installed atdd lives under .../site-packages/atdd — never source."""
+    _make_fake_consumer_repo(tmp_path)
+    fake_pkg = tmp_path / ".venv" / "lib" / "python3.14" / "site-packages" / "atdd"
+    (fake_pkg).mkdir(parents=True)
+    (fake_pkg / "__init__.py").write_text("")
+
+    fake_atdd = type("FakeAtdd", (), {"__file__": str(fake_pkg / "__init__.py")})
+
+    monkeypatch.chdir(tmp_path)
+    find_repo_root.cache_clear()
+    with patch.dict("sys.modules", {"atdd": fake_atdd}):
+        assert is_atdd_source_repo() is False
+
+
+def test_returns_false_when_pyproject_name_is_not_atdd(tmp_path, monkeypatch):
+    """Consumer repo whose pyproject.toml has a different name — not source."""
+    _make_fake_consumer_repo(tmp_path)
+    consumer_pkg = tmp_path / "src" / "atdd"
+    consumer_pkg.mkdir(parents=True)
+    (consumer_pkg / "__init__.py").write_text("")
+
+    fake_atdd = type("FakeAtdd", (), {"__file__": str(consumer_pkg / "__init__.py")})
+
+    monkeypatch.chdir(tmp_path)
+    find_repo_root.cache_clear()
+    with patch.dict("sys.modules", {"atdd": fake_atdd}):
+        assert is_atdd_source_repo() is False
+
+
+def test_returns_true_inside_actual_source_repo():
+    """When the real test is collected from the atdd source repo, True."""
+    find_repo_root.cache_clear()
+    assert is_atdd_source_repo() is True

--- a/src/atdd/coach/utils/repo.py
+++ b/src/atdd/coach/utils/repo.py
@@ -154,3 +154,62 @@ def require_repo_root(start: Optional[Path] = None) -> Path:
         f"No ATDD project markers found searching from {start_path}. "
         "Expected one of: .atdd/manifest.yaml, plan/ + contracts/, or .git/"
     )
+
+
+# Path parts that indicate a pip-installed / vendored location. Even if the
+# consumer's `.venv` sits inside their repo root, `atdd.__file__` under any of
+# these is a consumer install, NOT the atdd source repo.
+_VENDORED_PATH_MARKERS = frozenset(
+    {
+        "site-packages",
+        ".venv",
+        "venv",
+        ".tox",
+        "__pypackages__",
+        "node_modules",
+    }
+)
+
+
+def is_atdd_source_repo() -> bool:
+    """
+    Return True only when running inside the atdd source repo
+    (editable / source checkout), False in any consumer install.
+
+    Dogfood tests — tests that assert behavior against fixtures shipped
+    inside `src/atdd/**/validators/fixtures/` — must call this and
+    `pytest.skip(...)` when it returns False. Otherwise those tests leak
+    into consumer `atdd validate coder` runs and fail with assertion
+    errors against toolkit fixture data (see #272, #276).
+    """
+    try:
+        import atdd  # local import to avoid cycles at module import time
+
+        pkg_dir = Path(atdd.__file__).resolve().parent
+    except (ImportError, AttributeError, TypeError):
+        return False
+
+    if any(part in _VENDORED_PATH_MARKERS for part in pkg_dir.parts):
+        return False
+
+    try:
+        repo_root = find_repo_root().resolve()
+    except RuntimeError:
+        return False
+
+    try:
+        pkg_dir.relative_to(repo_root)
+    except ValueError:
+        return False
+
+    # Source repo always has a top-level pyproject.toml whose [project].name
+    # is "atdd" — the strongest signal that we are actually in the toolkit
+    # checkout and not in a consumer repo that happens to vendor atdd.
+    pyproject = repo_root / "pyproject.toml"
+    if not pyproject.is_file():
+        return False
+    try:
+        text = pyproject.read_text(encoding="utf-8")
+    except OSError:
+        return False
+    return 'name = "atdd"' in text

--- a/src/atdd/coder/validators/test_composition_completeness.py
+++ b/src/atdd/coder/validators/test_composition_completeness.py
@@ -24,7 +24,7 @@ import atdd
 import pytest
 import yaml
 
-from atdd.coach.utils.repo import find_repo_root
+from atdd.coach.utils.repo import find_repo_root, is_atdd_source_repo
 from atdd.coder.utils.python_file_walker import walk_consumer_python_files
 
 
@@ -698,6 +698,8 @@ def test_composition_completeness_python_fixture_passes_for_complete_and_partial
     When: Analyzing composition completeness
     Then: No violations are reported
     """
+    if not is_atdd_source_repo():
+        pytest.skip("fixture-based dogfood test — runs only inside the atdd source repo (#276)")
     violations = analyze_python_repo(FIXTURE_ROOT / "python_pass")
     assert_no_violations(violations)
 
@@ -711,6 +713,8 @@ def test_composition_completeness_python_fixture_detects_missing_setter_call():
     When: Analyzing composition completeness
     Then: The composition root is reported as missing presentation reachability
     """
+    if not is_atdd_source_repo():
+        pytest.skip("fixture-based dogfood test — runs only inside the atdd source repo (#276)")
     violations = analyze_python_repo(FIXTURE_ROOT / "python_fail_setter")
     assert len(violations) == 1, "expected one composition root violation"
     violation = violations[0]
@@ -728,6 +732,8 @@ def test_composition_completeness_typescript_fixture_detects_cameo_and_import_ty
     When: Analyzing composition completeness
     Then: Application layer violations are reported for both unwired hooks
     """
+    if not is_atdd_source_repo():
+        pytest.skip("fixture-based dogfood test — runs only inside the atdd source repo (#276)")
     violations = analyze_typescript_repo(FIXTURE_ROOT / "typescript_repo")
     violation_map = {(item.feature, item.file): item for item in violations}
 
@@ -750,6 +756,8 @@ def test_composition_completeness_typescript_fixture_accepts_barrels_and_partial
     When: Analyzing composition completeness
     Then: Barrel-consumed hooks pass and partial-feature integration is skipped
     """
+    if not is_atdd_source_repo():
+        pytest.skip("fixture-based dogfood test — runs only inside the atdd source repo (#276)")
     violations = analyze_typescript_repo(FIXTURE_ROOT / "typescript_repo")
     violated_files = {item.file for item in violations}
 

